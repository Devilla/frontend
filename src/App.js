import React from 'react';
import { browserHistory } from 'react-router';
import { AppContainer } from 'react-hot-loader';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import rootSaga from './sagas';
import reducer from './ducks';
import thunk from 'redux-thunk';
import { syncHistoryWithStore } from 'react-router-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import createSagaMiddleware from 'redux-saga';
import { StripeProvider } from 'react-stripe-elements';

const sagaMiddleware = createSagaMiddleware();

export const store = createStore(reducer, composeWithDevTools(
  applyMiddleware(sagaMiddleware, thunk)
));

const routerHistory = syncHistoryWithStore(browserHistory, store, {selectLocationState: state => state.get('router')});


sagaMiddleware.run(rootSaga);

const App = ({Component}) =>
  <AppContainer errorReporter={Error}>
    <Provider store={store}>
      <StripeProvider apiKey={process.env.NODE_ENV === 'production' ? process.env.REACT_APP_STRIPE_KEY : process.env.REACT_APP_DEVELOPMENT_STRIPE_KEY}>
        <Component routerHistory={routerHistory} />
      </StripeProvider>
    </Provider>
  </AppContainer>;

export default App;
