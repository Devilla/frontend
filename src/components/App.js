import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import  '../assets/stylesheets/slick.css'
import  '../assets/stylesheets/slick-theme.css'
import  '../assets/stylesheets/style.css'

import Header from './_common/Header';
import Footer from './_common/Footer';
import { Spinner } from './index.js';

class App extends Component {
  render(){
    return (
        <div className="app-container">
          <Header />

          <Spinner loading={this.props.loading} />
          <div className="content">
            {this.props.children}
          </div>
          <Footer />
        </div>
    );
  }
};

App.propTypes = {
  children: PropTypes.node
};

const mapStateToProps = state => ({
  loading: state.getIn(['loading', 'state'])
});

export default connect(mapStateToProps)(App);
