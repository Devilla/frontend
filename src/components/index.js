export { default as WebsiteHeader } from "./WebsiteHeader/WebsiteHeader";
export { default as WebsiteFooter } from "./WebsiteFooter/WebsiteFooter";
export { default as WebsiteHome } from "./WebsiteHome/WebsiteHome";
export { default as WebsiteHowItWorks } from "./WebsiteHowItWorks/WebsiteHowItWorks";
export { default as WebsiteIntegrations } from "./WebsiteIntegrations/WebsiteIntegrations";
export { default as WebsitePricing } from "./WebsitePricing/WebsitePricing";
export { default as WebsiteAbout } from "./WebsiteAbout/WebsiteAbout";
export { default as WebsiteSignIn } from "./WebsiteSignIn/WebsiteSignIn";
export { default as WebsiteSignUp } from "./WebsiteSignUp/WebsiteSignUp";
export { default as WebsiteContact } from "./WebsiteContact/WebsiteContact";
export { default as WebsiteTerms } from "./WebsiteTerms/WebsiteTerms";
export { default as WebsitePolicy } from "./WebsitePolicy/WebsitePolicy";
export { default as WebsiteDemoPage}  from "./WebsiteDemoPage/WebsiteDemoPage";
export { default as WebsiteCustStory}  from "./WebsiteCustStory/WebsiteCustStory";
export { default as WebsiteFeature}  from "./WebsiteFeature/WebsiteFeature";
export { default as Affiliate}  from "./Affiliate/Affiliate";
export { default as AffiliateRegister}  from "./Affiliate/AffiliateRegister";

export { default as Header } from "./Header/Header";
export { default as ForgetPassword } from "./ForgetPassword";
export { default as PricePage } from "./PricePage";
export { default as Common } from "./Common/function";
export { default as ResetPassword } from "./ResetPassword";
export { default as Sidebar } from "./Sidebar/Sidebar";
export { default as StatsCard } from "./StatsCard/StatsCard";
export { default as Dashboard } from './Dashboard/Dashboard';
export { default as Notification } from './Notification';
export { default as Analytics } from './Analytics/analytics';
export { default as AnalyticsProfile } from './Analytics/AnalyticsProfile';
export { default as New } from './New/new';
export { default as LoginFlow } from './LoginFlow/LoginFlow';
export { default as Spinner } from './Spinner/Spinner';
export { default as Error } from './Error/Error';
export { default as ComingSoon } from './ComingSoon/ComingSoon';
export { default as Button } from './Button/Button';
export { default as Profile } from './Profile/Profile';
export { default as BillingDetails } from './Billing/BillingDetails';
export { default as UpgradeCard } from './UpgradeCard/UpgradeCard';
export { default as UpgradePlan } from './UpgradePlan/UpgradePlan';
export { default as Integration } from './Integration/Integration';
export { default as Template } from './Template';
export { default as VerificationPage } from './VerificationPage/VerificationPage';
