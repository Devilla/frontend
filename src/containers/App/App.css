@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
}
