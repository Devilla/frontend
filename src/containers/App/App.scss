@import url('https://fonts.googleapis.com/css?family=Open+Sans:200,300,400,400i,500,600,700');
@import url('https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,300,400,400i,600,700');

body {
  font-family: "Raleway, Source Sans Pro, Verdana, Helvetica", sans-serif;
  display: block;
  margin: 0px;
}

.website-app {
  .alert-warning {
    background-color: #f2dede;
    border-color: #ebccd1;
    color: #a94442;
  }
  .content {
    padding-top: 100px;
  }
  .toast-style {
    padding: 17px 7px;
    background-color: rgba(0, 0, 0, 0.9) !important;
    color: #fff !important;
    text-align: center !important;
    height: 10px !important;
    border-radius: 5px !important;
    button {
      height: 1.5em;
      color: #fff;
    }
  }

  .transition-wrapper {
    position: relative;
    z-index: 1;
    .transition-item {
      position: fixed;
      top: 0;
      right: 0;
      left: 0;
    }
  }
  .signin-container,
  .authpage,
  .websitepolicy-container,
  .websitehowitworks-container,
  .websitehome-container,
  .WebsiteTerms-container,
  .websitepricing-container,
  .website-payment-container,
  .websiteintegrations-container,
  .demopage-container,
  .website-cust-story,
  .websitecontact-container,
  .websiteabout-container,
  .WebsiteFeature-container,
  .affiliateregister-container
   {
    overflow: hidden;
    box-sizing: border-box;
    padding: 20px;
    min-height: 100vh;
    transition: 500ms ease-out;

    &.transition-leave {
      opacity: 1;
    }

    &.transition-leave.transition-leave-active {
      opacity: 1;
      transform: translate3d(100%, 0, 0);
    }
  }
}
