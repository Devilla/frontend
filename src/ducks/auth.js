import { fromJS, List } from 'immutable';

const action = name => `/auth/${name}`;

export const FETCH = action('FETCH');
export const FETCHROLES = action('FETCHROLES');
export const LOGIN_SUCCESS = action('LOGIN_SUCCESS');
export const CHECK_TOKEN_EXISTS = action('CHECK_TOKEN_EXISTS');
export const FETCH_USER_SUCCESS = action('FETCH_USER_SUCCESS');
export const FETCH_ROLES_SUCCESS = action('FETCH_ROLES_SUCCESS');
export const UPDATE_USER = action('UPDATE_USER');


export const GDPR_FORM = action('GDPR_FORM');
export const GDPR_FORM_ERROR = action('GDPR_FORM_ERROR');

export const FORGOT_PASSWORD = action('FORGOT_PASSWORD');
export const FORGOT_PASSWORD_ERROR = action('FORGOT_PASSWORD_ERROR');
export const CLEAR_FORGOT_PASSWORD_ERROR = action('CLEAR_FORGOT_PASSWORD_ERROR');
export const SOCIAL_LOGIN = action('SOCIAL_LOGIN');
export const VERIFY_USER = action('VERIFY_USER');
export const VALIDATE_COUPON = action('VALIDATE_COUPON');
export const COUPON_SUCCESS = action('COUPON_SUCCESS');
export const COUPON_ERROR = action('COUPON_ERROR');
export const CLEAR_COUPON_ERROR = action('CLEAR_COUPON_ERROR');

export const AFFILIATE = action('AFFILIATE');
export const AFFILIATE_ERROR = action('AFFILIATE_ERROR');
export const CLEAR_AFFILIATE_ERROR = action('CLEAR_AFFILIATE_ERROR');

export const CONTACT_US = action('CONTACT_US');
export const CONTACT_ERROR = action('CONTACT_ERROR');
export const CLEAR_CONTACT_ERROR = action('CLEAR_CONTACT_ERROR');

export const DEMO = action('DEMO');
export const DEMO_ERROR = action('AFFILIATE_ERROR');
export const CLEAR_DEMO_ERROR = action('CLEAR_AFFILIATE_ERROR');

export const fetchUser = () => ({ type: FETCH });
export const fetchRoles = () => ({ type: FETCHROLES });
export const fetchUserSuccess = (user) => ({ type: FETCH_USER_SUCCESS, user });
export const updateUser = (user) => ({ type: UPDATE_USER, user });
export const fetchRolesSuccess = (roles) => ({ type: FETCH_ROLES_SUCCESS, roles });

export const affiliate = (data) => ({ type: AFFILIATE, data});
export const affiliateError = (data) => ({ type: AFFILIATE_ERROR, data });
export const clearAffiliateError = () => ({ type: CLEAR_AFFILIATE_ERROR });

export const contactUs = (data) => ({ type: CONTACT_US, data});
export const contactError = (data) => ({ type: CONTACT_ERROR, data });
export const clearContactError = () => ({ type: CLEAR_CONTACT_ERROR });

export const demo = (data) => ({ type: DEMO, data});
export const demoError = (data) => ({ type: DEMO_ERROR, data });
export const clearDemoError = () => ({ type: CLEAR_DEMO_ERROR });

export const gdprform = (data) => ({ type: GDPR_FORM, data });
export const gdprformError = (error) => ({ type: GDPR_FORM_ERROR, error });

export const forgotPassword = (data) => ({ type: FORGOT_PASSWORD, data });
export const forgotPasswordError = (error) => ({ type: FORGOT_PASSWORD_ERROR, error });
export const clearForgotPasswordError = () => ({ type: CLEAR_FORGOT_PASSWORD_ERROR });
export const socialLogin = (url) => ({ type: SOCIAL_LOGIN, url });
export const verifyUser = (code) => ({ type: VERIFY_USER, code });
export const validateCoupon = (coupon) => ({ type: VALIDATE_COUPON, coupon });
export const couponSuccess = (coupon) => ({ type: COUPON_SUCCESS, coupon });
export const couponError = (error) => ({ type: COUPON_ERROR, error });
export const clearCouponError = () => ({ type: CLEAR_COUPON_ERROR });

export const loginSuccess = (res) => ({ type: LOGIN_SUCCESS, res });
export const checkTokenExists = (token) => ({ type: CHECK_TOKEN_EXISTS, token });

const initialState = fromJS({
  user: {},
  roles: List([]),
  coupon: '',
  couponError: '',
  forgetError: '',
  affiliateError: '',
  gdprformError:''
});

const auth = (state = initialState, action) => {
  switch (action.type) {
    case LOGIN_SUCCESS:
      return state.set('user', action.res.user);
    case FETCH_USER_SUCCESS:
      return state.set('user', action.user);
    case FETCH_ROLES_SUCCESS:
      return state.set('roles', action.roles.roles);
    case COUPON_SUCCESS:
      return state.set('coupon', action.coupon).set('couponError', '');
    case COUPON_ERROR:
      return state.set('couponError', action.error);
    case CLEAR_COUPON_ERROR:
      return state.set('coupon', '').set('couponError', '');
    case FORGOT_PASSWORD_ERROR:
      return state.set('forgetError', action.error);
    case CLEAR_FORGOT_PASSWORD_ERROR:
      return state.set('forgetError', '');
    case AFFILIATE_ERROR:
      return state.set('affiliateError', action.error);
    case CLEAR_AFFILIATE_ERROR:
      return state.set('affiliateError', '');
    case CONTACT_ERROR:
      return state.set('contactError', action.error);
    case CLEAR_CONTACT_ERROR:
      return state.set('contactError', '');
    case DEMO_ERROR:
      return state.set('demoError', action.error);
    case CLEAR_DEMO_ERROR:
      return state.set('demoError', '');
    case GDPR_FORM_ERROR:
      return state.set('gdprformError', '');

    default:
      return state;
  }
};

export default auth;
