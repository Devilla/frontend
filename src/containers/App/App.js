import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { ToastContainer } from 'react-toastify';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { Spinner } from '../../components/index.js';
import { WebsiteHeader, WebsiteFooter } from 'components';

import { checkTokenExists } from 'ducks/auth';

import './scss/stack-interface.scss';
import './scss/socicon.css';
import './scss/iconsmind.scss';
import './scss/bootstrap.scss';
import './scss/theme.scss';
import './scss/custom.scss';
import './scss/font-raleway.scss';
import './scss/flickity.scss';
import './scss/font-sourcesanspro.scss';
import './App.scss';

class App extends Component {
  componentWillMount() {
    this.checkLogin((err) => {
      if(!err) {
        browserHistory.push('/dashboard');
      }
    });
  }

  checkLogin(callback) {
    const cookie = localStorage.getItem('authToken');
    const authToken = cookie
      ? JSON.parse(cookie)
      : null;
    if (authToken) {
      this.props.checkTokenExists(authToken);
      callback()
    } else {
      callback("not logged in")
    }
  }

  render(){
    return (
      <div className="website-app">
        <div className="basic-gradient-light" data-smooth-scroll-offset="77">
           <WebsiteHeader />
           <Spinner loading={this.props.loading} />
           <div className="content">
             {this.props.children}
           </div>
           <WebsiteFooter />
        </div>
        <ToastContainer hideProgressBar={true} />
      </div>
    );
  }

}

const mapDispatchToProps = {
  checkTokenExists
};

export default connect(null, mapDispatchToProps)(App);
