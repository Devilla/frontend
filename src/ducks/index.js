import { combineReducers } from 'redux-immutable';
import { routerReducer } from 'react-router-redux'

import profile from './profile';
import auth from './auth';
import plan from './plan';
import payment from './payment';
import notification from './notification';
import campaign from './campaign';
import rules from './rules';
import loading from './loading';
import configuration from './configuration';
import elastic from './elastic';
import pageurl from './pageurl';
import webhooks from './webhooks';
import subcampaign from './subcampaign';

const reducer = combineReducers({
  profile,
  auth,
  plan,
  payment,
  notification,
  campaign,
  rules,
  loading,
  configuration,
  elastic,
  pageurl,
  webhooks,
  subcampaign,
  router: routerReducer
});

export default reducer;
