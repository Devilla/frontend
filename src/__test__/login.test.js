import React from 'react';
import '../setupTests';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
// import Adapter from 'enzyme-adapter-react-15';


import { loginSuccess } from 'ducks/auth';
import {createStore} from 'redux'
import { load, loaded } from 'ducks/loading';
import { validateEmail, validatePassword, login, PASSWORD_MAX_LENGTH } from 'services/FormUtils';
import { WebsiteSignIn } from 'components';
import { store } from 'App';
import { base } from 'services/api';

// Snapshot for Home React Component
describe('>>>SignIn --- Snapshot',()=>{
  it('+++capturing Snapshot of SignIn', () => {
    const renderedValue =  renderer.create(<WebsiteSignIn store={store} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('check for in', ()=> {
  let wrapper;
  beforeEach ( () => {
       wrapper = shallow(<WebsiteSignIn  store={store}/>);
  })

  it('+++ render the DUMB component', () => {
    expect(wrapper.length).toEqual(1);
  });


  it('+++ contains website - error', () => {
    let wrapper = shallow(<WebsiteSignIn  store={store}/>);
    expect(wrapper.find('.website-error').length).toBe(0);
  });

  /*Not rendering components correctly*/

 //  it('should respond to change event and change the state of the WebsiteSignIn Component', () => {
 //   wrapper.find('input').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});
 //   expect(wrapper.state('email')).toEqual('blah@gmail.com');
 // });

    // it('should respond to change event and change the state of the WebsiteSignIn Component', () => {
    //  wrapper.find('input#email').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});
    //  expect(wrapper.state('email')).toEqual('blah@gmail.com');
    // })

    // it('should respond to changed  event and change the state of the SignIn Component', () => {
    //  wrapper.find('input#password').simulate('change', {target: {name: 'password', value: 'cats'}});
    //  expect(wrapper.state('password')).toEqual('cats');
    // })

})
