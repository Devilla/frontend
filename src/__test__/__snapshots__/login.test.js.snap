// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>H O M E --- Snapshot +++capturing Snapshot of Home 1`] = `
<div
  className="main-container"
>
  <section
    className="switchable switchable--switch bg--secondary"
  >
    <div
      className="container"
    >
      <div
        className="row justify-content-center"
      >
        <div
          className="col-sm-12 col-md-7 col-lg-5"
        >
          <div
            className=""
          >
            <h2>
              Welcome back!
            </h2>
            <p
              className="lead"
            >
              <span>
                Don't have an account yet?Â 
                <a
                  onClick={[Function]}
                  style={Object {}}
                >
                  Sign up
                </a>
              </span>
            </p>
            <hr
              className="short"
            />
            <form
              className="loginfrm"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                
                <div
                  className="col-9 "
                >
                  <input
                    className="fieldname"
                    id="emails"
                    name="email"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="email"
                  />
                  <span
                    className="help-block"
                  >
                    <p
                      className="website-error"
                    >
                      
                    </p>
                  </span>
                </div>
                <div
                  className="col-9"
                >
                  <input
                    className="field w-input "
                    id="password"
                    maxLength={20}
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                  />
                  <span
                    className="help-block"
                  >
                    <p
                      className="website-error"
                    >
                      
                    </p>
                  </span>
                </div>
                <div
                  className="col-9 frmcntl"
                >
                  <input
                    className="button submit-button w-button btn btn--primary ml-0"
                    disabled={true}
                    id="subtract"
                    style={
                      Object {
                        "cursor": "pointer",
                      }
                    }
                    type="submit"
                    value="Login"
                  />
                </div>
                <div>
                  <a
                    onClick={[Function]}
                    style={
                      Object {
                        "padding": "25px",
                      }
                    }
                  >
                    Forgot password?
                  </a>
                </div>
              </div>
            </form>
          </div>
        </div>
        <div
          className="vristrue"
        />
        <div
          className="col-md-4 col-lg-4"
        >
          <p>
             
            <br />
          </p>
          <div
            className="mt--2"
          >
            <a
              className="link "
              href="undefinedconnect/facebook"
            >
              <a
                className="btn btn--icon bg--facebook"
                onClick={[Function]}
                style={Object {}}
              >
                <span
                  className="btn__text "
                >
                  <i
                    className="socicon socicon-facebook"
                  />
                  Login with Facebook
                </span>
              </a>
            </a>
            <p />
            <a
              className="link"
              href="undefinedconnect/google"
            >
              <a
                className="btn btn--icon bg--googleplus"
                onClick={[Function]}
                style={Object {}}
              >
                <span
                  className="btn__text"
                >
                  <i
                    className="socicon socicon-google"
                  />
                  Login with Google
                </span>
              </a>
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
